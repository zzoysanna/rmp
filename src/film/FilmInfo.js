import React, { Component } from 'react';
import './styles/film-info.less';
import { connect } from "react-redux";

class FilmInfo extends Component {

	prepareDate(date) {
		const releaseDate = new Date(date);
		return releaseDate.getFullYear();
	}

	render() {
		const {
			title,
			release_date,
			genres,
			poster_path,
			overview,
			runtime,
			vote_average
		} = this.props.film;
		const {id, film} = this.props;
		return (
			<div className="info">
				<div style={{ backgroundImage: `url(${poster_path})` }} className="info-image">&nbsp;</div>
				<div className="info-description">
					<div className="info-header">
						<h1>{title}</h1>
						<span className="info-rating">{vote_average}</span>
					</div>
					<p className="info-genre">{genres.join('&')}</p>
					<div className="info-timing">
						<p><span>{this.prepareDate(release_date)}</span>year</p>
						<p><span>{runtime}</span>min</p>
					</div>
					<div>{overview}</div>
				</div>
			</div>
		);
	}
}

const mapStateToProps = (state, ownProps) => {
	return {
		film: state.currentFilm,
		id: ownProps.match.params.id
	}
};

const mapDispatchToProps = (dispatch) => {
	return {}
};

export default connect(mapStateToProps, mapDispatchToProps)(FilmInfo);
