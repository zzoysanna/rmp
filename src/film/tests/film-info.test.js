import React from 'react';
import { shallow } from 'enzyme';
import FilmInfo from '../FilmInfo';

describe('FilmInfo', () => {
	let film;

	beforeAll(() => {
		film = {
			id: 1,
			title: 'test-title',
			poster_path: 'test-poster.jpg',
			genres: ['a', 'b'],
			release_date: '1997',
			overview: 'test-description',
			runtime: 157,
			vote_average: 4.5,
		};
	});

	it('should be render correctly', () => {
		const component = shallow(<FilmInfo film={film}/>);
		expect(component).toMatchSnapshot();
	});
});
