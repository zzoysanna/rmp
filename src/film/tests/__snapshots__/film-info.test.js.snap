// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmInfo should be render correctly 1`] = `ShallowWrapper {}`;
